src/core/gui/dialog/FormatDialog.cpp

```diff
@@ -137,8 +101,11 @@ void FormatDialog::spinValueChangedCb(FormatDialog* dlg) {
 
     int i = 0;
     for (auto& size: dlg->paperSizes) {
-        double w = gtk_paper_size_get_width(size.get(), GTK_UNIT_POINTS);
-        double h = gtk_paper_size_get_height(size.get(), GTK_UNIT_POINTS);
+        xoj_assert(std::holds_alternative<PaperFormatUtils::GtkPaperSizeUniquePtr_t>(size));
```

Comment:
At this point in the code it is not necessary anymore. I opted for this implementation because it avoids creating another vector by continuing to use the vector filled from line 49 to 53 (https://github.com/Chrisimx/xournalpp/blob/e37485545473bab0bd73e373188af1fe69fd8cd9/src/core/gui/dialog/FormatDialog.cpp#L49-L53) which has to hold std::string / GtkPaperSizeUPtr variants to represent the custom option (this is shortly after removed from the vector so that only GtkPaperSizeUPtrs are in the vector). You could then move everything into a new vector which just holds GtkPaperSizeUPtrs but I'm not sure. It would make it possible to remove the xoj_assert, yet it would also make it necessary to create another vector and check and move each element from the old one, or (as another option) also create a second vector but then call PaperFormatUtils::loadDefaultPaperSizes a second time. In this case, you'd have to update the PaperFormatUtils::loadDefaultPaperSizes function to support putting the items into a vector which uses GtkPaperSizeUPtrs instead of a std::string / GtkPaperSizeUPtr variant.